test/e2e/operator.go

```diff
@@ -508,3 +509,70 @@ var _ = Describe("ARO Operator - ImageConfig Reconciler", func() {
 		}).WithContext(ctx).Should(Succeed())
 	})
 })
+
+var _ = Describe("ARO Operator - dnsmasq", func() {
+	const (
+		timeout = 1 * time.Minute
+		polling = 10 * time.Second
+	)
+	mcpName := "test-aro-custom-mcp"
+	mcName := fmt.Sprintf("99-%s-aro-dns", mcpName)
+
+	ctx := context.Background()
+
+	customMcp := mcv1.MachineConfigPool{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "machineconfiguration.openshift.io/v1",
+			Kind:       "MachineConfigPool",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: mcpName,
+		},
+		Spec: mcv1.MachineConfigPoolSpec{},
+	}
+
+	getMachineConfigNames := func(g Gomega) []string {
+		machineConfigs, err := clients.MachineConfig.MachineconfigurationV1().MachineConfigs().List(ctx, metav1.ListOptions{})
```

Comment:
If we define `getMachineConfigNames` like this - we capture `ctx` from the `Describe` node. We should be using context from `Eventually`.